compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2562,6 +2633,11 @@ fn add_upstream_rust_crates<'a>(
                     let skip_because_lto =
                         upstream_rust_objects_already_included && is_rust_object && is_builtins;
 
+                    // if sess.opts.unstable_opts.separate_native_rlib_dependencies &&
+                    if f.ends_with(".a") {
```

Comment:
This needs a comment telling what is this file and why are we removing it.
(Also commented out code needs to be removed.)

Reply:
We are removing it 1) as an optimization (it doesn't participate in linking at this point, so no need to copy garbage around) and 2) because we don't want it to end up as a single remaining entry in the rlib archive (then some linkers may complain).